railties/test/generators/plugin_generator_test.rb

```diff
@@ -75,6 +75,23 @@ def test_generating_without_options
     assert_no_file "bin/rails"
   end
 
+  def test_generating_in_full_mode_without_options
+    run_generator [destination_root, "--full"]
+    assert_file "bin/rails", /\s+require\s+["']rails\/all["']/
```

Comment:
There is already a test to generate full mode engine. I think that it is better to add it there.
https://github.com/rails/rails/blob/df94b863c2ff8f1bcf12e36ed8fc1419292668e7/railties/test/generators/plugin_generator_test.rb#L270-L284